### PR TITLE
feat: integrate Android native module and refine module spec

### PR DESCRIPTION
**Pull request by gmemmy**
## 📝 Description

A clear and concise description of what this pull request does.

## 🎯 Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [x] 🧹 Code refactoring
- [x] ⚡ Performance improvement
- [ ] 🔒 Security fix
- [x] 🧪 Test addition or update

## 🔗 Related Issue

Closes #[issue-number]

### Test Coverage

- [x] Unit tests added/updated
- [x] Integration tests added/updated
- [x] Manual testing completed

### Platforms Tested

- [ ] iOS
- [ ] Android
- [x] Both

### Test Scenarios

- [x] Biometric authentication
- [ ] Secure storage operations
- [ ] PIN authentication
- [ ] Digital signing
- [ ] React hooks
- [ ] Error handling
- [ ] Edge cases

### Code Quality

- [ ] My code follows the project's style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### Documentation

- [x] README.md updated (if applicable)
- [x] API documentation updated (if applicable)
- [ ] Code comments added/updated
- [ ] Changelog updated (if applicable)

### Build & CI

- [x] All CI checks pass
- [x] Build succeeds locally
- [ ] No TypeScript errors
- [ ] No linting errors
- [ ] No formatting issues

## 🚀 Breaking Changes

If this PR includes breaking changes, please describe them and provide migration instructions:

### What Changed

- [x] API signature changes
- [x] Configuration changes
- [ ] Behavior changes
- [ ] Platform requirements changes

### Migration Guide

```typescript
// Before
import { oldFunction } from '@gmemmy/react-native-biolink';
const result = oldFunction(param1, param2);

// After
import { newFunction } from '@gmemmy/react-native-biolink';
const result = newFunction({ param1, param2 });
```

### iOS

- [ ] Native code changes
- [ ] Podfile changes
- [ ] Entitlements changes
- [ ] Info.plist changes

### Android

- [ ] Native code changes
- [ ] build.gradle changes
- [ ] AndroidManifest.xml changes
- [ ] ProGuard rules changes

## 🔍 Additional Notes

Any additional information that reviewers should know:

- Performance implications
- Security considerations
- Backward compatibility notes
- Known limitations
- Future considerations

## 📸 Screenshots/Videos

If applicable, add screenshots or videos to demonstrate the changes.

---

**Note**: Please ensure all tests pass and the code follows the project's contribution guidelines. Thank you for your contribution! 🎉

## Summary by CodeRabbit

* **New Features**
  * Added Android biometric, secure storage, and hardware-backed signing integration with native support and improved lifecycle handling.
  * New demo scripts for iOS simulator, pod install, and CI workflow.

* **Bug Fixes**
  * Replaced native blur UI with semi-transparent overlays to remove a native blur dependency.
  * Adjusted SDK/tool versions for platform stability.

* **Documentation**
  * Rewrote main and demo READMEs; clarified API usage, types, and quick links.

* **Chores**
  * Updated CI/package scripts, lint/format ignore files, and repo ignore patterns.

* **Refactor**
  * Large Android module restructuring, replacing legacy implementation with a new native module architecture.

### CODE REVIEW

File: packages/react-native-biolink/android/src/main/kotlin/com/margelo/nitro/biolink/HybridBiolinkCore.kt [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            } else {
+                Log.i(TAG, "Secret retrieved successfully for key: $key value=$value")
+                promise.resolve(value)
+            }
+        } catch (e: Exception) {
+            Log.e(TAG, "Failed to get secret for key: $key", e)
+            promise.reject(RuntimeException("Failed to get secret: ${e.message}", e))
+        }
+        return promise
+    }
```
[the remark this answers is not in the document]
Comment (gmemmy) [reply packages/react-native-biolink/android/src/main/kotlin/com/margelo/nitro/biolink/HybridBiolinkCore.kt:129]: There's a bug with how nitro's Promise API handles null values and how that is propagated to JS. I kept running into an NPE of some sort until I decided to return an empty string since on the JS side we can account for that
